services/web/server/src/simcore_service_webserver/meta_handlers.py

```diff
@@ -0,0 +1,277 @@
+    # Select range on those tagged as iterations and returned their assigned workcopy id
+    # Can also check all associated project uuids and see if they exists
+    # FIXME: do all these operations in database
+    tags_per_child: List[List[TagProxy]] = await vc_repo.get_children_tags(
+        repo_id, commit_id
+    )
+
+    iterations: List[Tuple[ProjectID, CommitID]] = []
+    for n, tags in enumerate(tags_per_child):
+
+        # FIXME: the db query did not guarantee
+        # not sorted
+        # not limited
+        # not certain if tags we need
+        try:
+            iteration: Optional[ProjectIteration] = None
+            workcopy_id: Optional[ProjectID] = None
+
+            for tag in tags:
+                if pim := ProjectIteration.from_tag_name(
+                    tag.name, return_none_if_fails=True
+                ):
+                    if iteration:
+                        raise ValueError(
+                            f"This entry has more than one iteration {tag=}"
+                        )
+                    iteration = pim
+                elif pid := parse_workcopy_project_tag_name(tag.name):
+                    if workcopy_id:
+                        raise ValueError(
+                            f"This entry has more than one workcopy  {tag=}"
+                        )
+                    workcopy_id = pid
+                else:
+                    log.debug("Got %s for children of %s", f"{tag=}", f"{commit_id=}")
+
+            if not workcopy_id:
+                raise ValueError("No working copy found")
+            if not iteration:
+                raise ValueError("No iteration tag found")
+
+            iterations.append((workcopy_id, iteration.iter_index))
+
+        except ValueError as err:
+            log.debug("Skipping child %s: %s", n, err)
+
+    total_number_of_iterations = len(iterations)
+
+    # sort and select. If requested interval is outside of range, it returns empty
+    iterations.sort(key=lambda tup: tup[1])
+
+    if limit is None:
+        return iterations[offset:], total_number_of_iterations
+
+    return iterations[offset : (offset + limit)], total_number_of_iterations
+
+
+async def create_or_get_project_iterations(
+    vc_repo: VersionControlForMetaModeling,
+    project_uuid: ProjectID,
+    commit_id: CommitID,
+) -> List[IterationTuple]:
+
+    raise NotImplementedError()
+
+
+# MODELS ------------------------------------------------------------
+
+
+class ParentMetaProjectRef(BaseModel):
+    project_id: ProjectID
+    ref_id: CheckpointID
+
+
+class ProjectIterationAsItem(BaseModel):
+    name: str = Field(
+        ...,
+        description="Iteration's resource name [AIP-122](https://google.aip.dev/122)",
+    )
+    parent: ParentMetaProjectRef = Field(
+        ..., description="Reference to the the meta-project that created this iteration"
+    )
+
+    workcopy_project_id: ProjectID = Field(
```

Comment:
very minor: work vs working it's only 3 letters ;)

Reply:
it was ``wcopy`` -> ``workcopy`` after @GitHK  review 
I think it is a good compromise ;-) since it is understandable